packages/data-clients/leonardo/src/LeoDisksV2DataClient.ts

```diff
@@ -0,0 +1,24 @@
+import { AbortOption, FetchFn } from '@terra-ui-packages/data-client-core';
+
+export interface LeoDisksV2DataClient {
+  delete: (diskId: number, options?: AbortOption) => Promise<void>;
+}
+const diskV2Root = 'api/v2/disks';
+
+export const makeLeoDisksV2DataClient = (deps: LeoDisksV2DataClientDeps): LeoDisksV2DataClient => {
```

Comment:
This is definitely not part of this PR, but I am still a bit confused at the huge difference between the use of the V1 vs V2 disk API in the UI, don't we need to list V2 disks as well? Or is the V1 list combining V1 and V2 resources already?

Reply:
we don't manage v2 disks directly beyond deleting, all other operations are done in the context of a VM (i.e. creating a disk with a VM). 

The list endpoint already lists all disks, since there's an option to not pass in googleProject as a filter